runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -816,8 +826,20 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       // Put the request back into the pool
       setMethodBeingCompiled(NULL); // Must have the compQmonitor
 
+
+      // Save the pointer to the plan before recycling the entry
+      TR_OptimizationPlan *optPlan = entry._optimizationPlan;
+      // This deallocates client options, so need to use a per-client allocator
       compInfo->requeueOutOfProcessEntry(&entry);
 
+      exitPerClientAllocationRegion();
+      if (optPlan)
+         {
+         TR_OptimizationPlan::freeOptimizationPlan(optPlan);
+         }
```

Comment:
I meant this is what I need to do to correctly free the optimization plan.
Moving client options outside of `MethodToBeCompiled` sounds like a better idea though